internal/cluster/cluster.go

```diff
@@ -645,11 +645,13 @@ func (m *Manager) VerifyClusterUpdatability(c *common.Cluster) (err error) {
 func (m *Manager) CancelInstallation(ctx context.Context, c *common.Cluster, reason string, db *gorm.DB) *common.ApiErrorResponse {
 	eventSeverity := models.EventSeverityInfo
 	eventInfo := "Cancelled cluster installation"
+	lastState := newStateCluster(c)
 	defer func() {
 		m.eventsHandler.AddEvent(ctx, *c.ID, nil, eventSeverity, eventInfo, time.Now())
+		m.metricAPI.ClusterInstallationFinished(ctx, "canceled", lastState.srcState, c.OpenshiftVersion, *c.ID, c.EmailDomain, c.InstallStartedAt)
```

Comment:
Should be a function to be used on both places

Reply:
Those 2 cases (The user facing event and the metrics) are not really the same - they are bundled here by chance. Such function will have too long parameters and will not be re-used too much. For 2 line defer this style is clearer